### PR TITLE
refactor(LinearAlgebra/BilinearForm/Basic): Deprecate

### PR DESCRIPTION
**Pull request by mans0954**
Deprecate most of `LinearAlgebra/BilinearForm/Basic`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — COMMENTED** [review]
I'd argue that a lot of these might be worth keeping; the `₂` naming is a little odd if you're working with bilinear forms.

The `coe` lemmas are not, as their statement is nonsense after the last PR.

File: Mathlib/LinearAlgebra/BilinearForm/Basic.lean [hunk 1]
```diff
@@ -164,62 +176,76 @@ theorem add_apply (x y : M) : (B + D) x y = B x y + D x y :=
...
+@[deprecated]
 instance {α} [Monoid α] [DistribMulAction α R] [DistribMulAction αᵐᵒᵖ R]
     [SMulCommClass R α R] [IsCentralScalar α R] :
     IsCentralScalar α (BilinForm R M) := inferInstance
 
+@[deprecated]
 instance : AddCommMonoid (BilinForm R M) := inferInstance
 
+@[deprecated]
 instance : Neg (BilinForm R₁ M₁) := inferInstance
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/BilinearForm/Basic.lean:206]: Deprecating instances is probably not a good idea; I think these should just be deleted.

**Comment by mans0954** [comment]
> I'd argue that a lot of these might be worth keeping; the `₂` naming is a little odd if you're working with bilinear forms.

I did wonder about that. We've ended up with some things in `LinearMap.BilinForm` and some things with the `₂` convention.

I suspect a lot of these things should be `LinearMap.BilinMap` or even `LinearMap.SesquiMap` so neither of the current naming conventions may be the final form?

**Comment by eric-wieser** [comment]
> I suspect a lot of these things should be LinearMap.BilinMap or even LinearMap.SesquiMap so neither of the current naming conventions may be the final form?

I would be inclined not to deprecate these until we have a clearer picture of the final form. Perhaps let's create a new PR that deprecates just the nonsense `coe` lemmas, and leave the rest until your `BilinMap` stuff lands?

**Comment by mans0954** [comment]
> I would be inclined not to deprecate these until we have a clearer picture of the final form. Perhaps let's create a new PR that deprecates just the nonsense `coe` lemmas, and leave the this PR until your `BilinMap` stuff lands?

Okay, what about the `_apply` theorems?

File: Mathlib/LinearAlgebra/BilinearForm/Basic.lean (hunk 1 above)
Comment (mans0954) [reply Mathlib/LinearAlgebra/BilinearForm/Basic.lean:206]: #12122

**Comment by eric-wieser** [comment]
The `apply` theorems are good and we should keep them.

**Comment by mans0954** [comment]
> Perhaps let's create a new PR that deprecates just the nonsense `coe` lemmas

#12132

**Comment by mans0954** [comment]
I'm going to close this PR for now in favour of:

- #12122
- #12124
- #12132

**Pull request closed without merging**